src/upgradeExecutorPrepareAddExecutorTransactionRequest.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Address,
+  PublicClient,
+  encodeFunctionData,
+  PrepareTransactionRequestReturnType,
+} from 'viem';
+import { upgradeExecutor } from './contracts';
+import {
+  UPGRADE_EXECUTOR_ROLE_EXECUTOR,
+  upgradeExecutorEncodeFunctionData,
+} from './upgradeExecutorEncodeFunctionData';
+
+/**
+ * Type for the params of the upgradeExecutorPrepareAddExecutorTransactionRequest function
+ */
+export type UpgradeExecutorPrepareAddExecutorTransactionRequestParams = {
+  account: Address;
+  upgradeExecutorAddress: Address;
+  executorAccountAddress: Address;
+  publicClient: PublicClient;
+};
+
+/**
+ * Prepares a transaction to grant the executor role to a new account
+ *
+ * - Example: [Add new executor account to UpgradeExecutor](https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/examples/upgrade-executor-add-account/index.ts)
```

Comment:
I think we can use Link instead here?

Reply:
What do you mean? "Add new executor account to UpgradeExecutor" should be a link. I follow the same structure as in `createRollup`.